### PR TITLE
Http strategy for plugging in other strategy apis.

### PR DESCRIPTION
**Pull request by pkehrer**
* Refactored tournament>match>game>player flow to use promises.
* added HttpProxyStrategy for adding players over an API.
* tweaked the logging to show some mores stuff
* Followed linus torvalds' git message formatting

### CODE REVIEW

File: model/game/game.js [hunk 1]
```diff
@@ -34,14 +36,15 @@ class Game {
...
-    let winner;
+    return this.deal().then(() => this.keepPlayingUntilWinner(this.playRound()));
+  }
 
-    // while there are cards and players remain, keep playing
-    while (!(winner = this.declareWinner())) {
-      this.playRound();
-    }
-    this.winner = winner;
+  keepPlayingUntilWinner(roundResult) {
```
Comment (pkehrer) [inline model/game/game.js:42]: I used this approach in both the match and the game.  Do you know a better pattern for times when you need to chain promises while waiting on a condition like it were a while loop?

File: model/game/tournament.js [hunk 2]
```diff
@@ -3,27 +3,39 @@ let Match = require("./match");
...
   }
 
   listWinners() {
     return _.map(this.matches, match => match.winner());
   }
 
   report() {
-    return _.map(_.groupBy(this.listWinners()), (value, key) => {
-      return { playerId: key, wins: value.length / this.numberOfMatches };
+    let report = _.map(_.groupBy(this.listWinners()), (value, key) => {
```
Comment (pkehrer) [inline model/game/tournament.js:30]: i wanted to see game and match counts as well as totals... not sure why.

File: strategy/httpproxy.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+          uri: this.uri,
+          method: "POST",
+          json: true,
+          body: { player, opponents }
+      });
+  }
+}
+
+
+module.exports = HttpProxyStrategy;
```
Comment (pkehrer) [inline strategy/httpproxy.js:19]: This file felt very node-y to me.

**Comment by pkehrer** [comment]
Oh, I should probably mention that this does run successfully.  I'm not 100% on my js style, but it works correctly as far as I know.

File: model/game/game.js (hunk 1 above)
Comment (daward) [reply model/game/game.js:42]: This seems fine to me.  We haven't actually had a cause to do any while-promises in any of our code at work.

File: model/game/tournament.js (hunk 2 above)
Comment (daward) [reply model/game/tournament.js:30]: I think it will be interesting.  We'll probably want to know if the matches are close.

File: index.js [hunk 4]
```diff
@@ -4,17 +4,24 @@ let noprincess = require("./strategy/noprincess");
...
+playerapi.start(true); // remove the parameter or set it to false to disable logging.
 
-let strategies = [];
-strategies.push(randohandmaid);
-_.times(3, () => strategies.push(noprincess));
-//strategies.push(noprincess);
+var httpStrategy = new HttpProxyStrategy("http://localhost:8080/api/takeTurn");
 
-let tournament = new Tournament(strategies, 1000);
+let strategies = [
```
Comment (daward) [inline index.js:14]: Yeah, thats a lot better.

**Pull request merged**